CONTRIBUTING.md

```diff
@@ -0,0 +1,388 @@
+Communication Channels
+----------------------
+
+Most communication about Gridcoin Research development happens on Slack, in the
+`#development` channel on TeamGridcoin. The invite link is in 
+[the main README](README.md#community).
+
+Discussion about code base improvements happens in GitHub issues and on pull
+requests.
+
+
+Contributor Workflow
+--------------------
+
+The codebase is maintained using the "contributor workflow" where everyone
+without exception contributes patch proposals using "pull requests". This
+facilitates social contribution, easy testing and peer review.
+
+To contribute a patch, the workflow is as follows:
+
+  1. Fork repository ([only for the first time](https://help.github.com/en/articles/fork-a-repo))
+  1. Create topic branch
+  1. Commit patches
+
+The project coding conventions in the [coding.txt](doc/coding.txt)
+must be followed.
+
+
+### Committing Patches
+
+In general, [commits should be atomic](https://en.wikipedia.org/wiki/Atomic_commit#Atomic_commit_convention)
+and diffs should be easy to read. For this reason, do not mix any formatting
+fixes or code moves with actual code changes.
+
+Commit messages should be verbose by default consisting of a short subject line
+(50 chars max), a blank line and detailed explanatory text as separate
+paragraph(s), unless the title alone is self-explanatory (like "Corrected typo
+in init.cpp") in which case a single title line is sufficient. Commit messages should be
+helpful to people reading your code in the future, so explain the reasoning for
+your decisions. Further explanation [here](https://chris.beams.io/posts/git-commit/).
+
+If a particular commit references another issue, please add the reference. For
+example: `refs #1234` or `fixes #4321`. Using the `fixes` or `closes` keywords
+will cause the corresponding issue to be closed when the pull request is merged.
+
+Commit messages should never contain any `@` mentions.
+
+Please refer to the [Git manual](https://git-scm.com/doc) for more information
+about Git.
+
+  - Push changes to your fork
+  - Create pull request
+
+
+### Creating the Pull Request
+
+The title of the pull request should be prefixed by the component or area that
+the pull request affects. Valid areas as:
+
+  - `consensus` for changes to consensus critical code
+  - `doc` for changes to the documentation
+  - `qt` or `gui` for changes to gridcoinresearch
+  - `log` for changes to log messages
+  - `mining` for changes to the mining code
+  - `net` or `p2p` for changes to the peer-to-peer network code
+  - `refactor` for structural changes that do not change behavior
+  - `rpc` for changes to the RPC
+  - `scraper` for changes to the Scraper
+  - `script` for changes to the scripts and tools
+  - `test`, `qa` or `ci` for changes to the unit tests, QA tests or CI code
+  - `util` or `lib` for changes to the utils or libraries
+  - `wallet` for changes to the wallet code
+  - `build` for changes to the GNU Autotools or reproducible builds
```

Comment:
We may need to add categories here appropriate to gridcoin, such as neuralnet (boy do I hate that name).

Reply:
Added `neuralnet`, what else should be added?

Reply:
I can't think of any more at the moment.